src/main.rs

```diff
@@ -82,10 +119,43 @@ impl LanguageServer for Backend {
         ])))
     }
 
-    async fn hover(&self, _: HoverParams) -> Result<Option<Hover>> {
+    async fn hover(&self, params: HoverParams) -> Result<Option<Hover>> {
+        let world = self.world.read().await;
+        let world = world.as_ref().unwrap();
+        let source = world.main();
```

Comment:
We should get the current document from `params`; the current `main` thing is a hack and will eventually be removed.